File: spark2-test/src/main/scala/com/lsy/myhadoop/spark2/beifeng/sql/commons/ParquetLoadData.scala
package com.lsy.myhadoop.spark2.beifeng.sql.commons

import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.sql.SQLContext

/**
 * @author Administrator
 */
object ParquetLoadData {
  
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
        .setAppName("ParquetLoadData")
        .setMaster("local")
    val sc = new SparkContext(conf)
    val sqlContext = new SQLContext(sc)
    
    val usersDF = sqlContext.read.parquet("hdfs://192.168.190.132:9000/test/users.parquet")
    usersDF.registerTempTable("users")
    val userNamesDF = sqlContext.sql("select name from users")  
    userNamesDF.rdd.map { row => "Name: " + row(0) }.collect()
        .foreach { userName => println(userName) }   
  }
  
}

File: spark2-test/src/main/java/com/lsy/myhadoop/sparkjava/sql/commons/ParquetLoadData.java
package com.lsy.myhadoop.sparkjava.sql.commons;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SQLContext;

import java.util.List;

/**
 * Created by root on 4/5/16.
 */
public class ParquetLoadData {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
                .setAppName("DataFrameCreate")
                .setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);
        SQLContext sqlContext = new SQLContext(sc);

        DataFrame df = sqlContext.read().load("hdfs://192.168.190.132:9000/test/users.parquet");

        df.registerTempTable("users");
        DataFrame userNameDF = sqlContext.sql("select name from users");

        List<String> userNames = userNameDF.javaRDD().map(new Function<Row, String>() {
            @Override
            public String call(Row row) throws Exception {
                return "Name: "+row.getString(0);
            }
        }).collect();

        for(String userName : userNames){
            System.out.println(userName);
        }

    }
}
